zebra-network/src/peer_set/set.rs

```diff
@@ -496,7 +507,18 @@ where
                     // - always do the same checks on every ready peer, and
                     // - check for any errors that happened right after the handshake
                     trace!(?key, "got Change::Insert from Discover");
-                    self.remove(&key);
```

Comment:
I just realised my suggestion here is wrong, we still have to do something about duplicate connections to the same IP and port.

So I opened #6983.

Reply:
> for outbound connections, this can cause connection thrashing, if two peers disagree on which of their connections is "latest"

Good thinking keeping the older connection.